src/engine/game/battle/enemybattler.lua

```diff
@@ -618,10 +621,31 @@ function EnemyBattler:defeat(reason, violent)
 
     if violent then
         Game.battle.used_violence = true
+        if self.recruit > 0 and self:getFlag("recruit", 0) ~= false then
+            if Game:getConfig("enableRecruits") and self.done_state ~= "FROZEN" then
+                self:recruitMessage("lost")
+            end
+            self:setFlag("recruit", false)
+        end
+        if self.done_state == "KILLED" or self.done_state == "FROZEN" then
+            Game.battle.xp = Game.battle.xp + self.experience
+        end
     end
-
+    
+    if self.recruit > 0 and type(self:getFlag("recruit", 0)) == "number" and (self.done_state == "PACIFIED" or self.done_state == "SPARED") then
+        self:addFlag("recruit", 1)
+        if Game:getConfig("enableRecruits") then
+            local counter = self:recruitMessage("recruit")
+            counter.first_number = self:getFlag("recruit", 0)
+            counter.second_number = self.recruit
+            Assets.playSound("mercyadd", 4.8, 0.8) -- Is this the sounds that plays when an enemy gets recruited? I have no idea.
+        end
+        if self:getFlag("recruit", 0) >= self.recruit then
+            self:setFlag("recruit", true)
+        end
+    end
+    
     Game.battle.money = Game.battle.money + self.money
-    Game.battle.xp = Game.battle.xp + self.experience
```

Comment:
While experience is an unused mechanic, it does get added here, and not on violence defeats.

Reply:
Ralsei talked in Chapter 1 about sparing to not get EXP, if this mechanic ever gets used, it's probably if you cause a fatal kill. I don't think it matters that much for now.

Reply:
I'd rather not go off of guesses and just use what's currently in Deltarune. We can always change it later.